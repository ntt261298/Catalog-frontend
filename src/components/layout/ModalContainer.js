import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import Loadable from 'react-loadable';

const LoginModal = Loadable({
  loader: () => import('./LoginModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
const SignupModal = Loadable({
  loader: () => import('./SignupModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});

const ModalContainer = (props) => {
  switch (props.modal) {
    case 'login':
      return <LoginModal />;
    case 'signup':
      return <SignupModal />;
    default:
      return null;
  }
};

ModalContainer.propTypes = {
  modal: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  modal: state.app.modal,
});
export default connect(mapStateToProps, null)(ModalContainer);
