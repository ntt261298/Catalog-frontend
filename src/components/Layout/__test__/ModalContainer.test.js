import React from 'react';
import { shallow } from '../../../enzyme';
import { ModalContainer } from '../ModalContainer';

describe('src/components/Layout/ModalContainer.js', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(
      <ModalContainer {...props} />,
    );
  };

  beforeEach(() => {
    props = {
      modal: 'login',
    };
  });

  it('should render login modal', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should render signup modal', () => {
    props.modal = 'signup';
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should render addItem modal', () => {
    props.modal = 'addItem';
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should render editItem modal', () => {
    props.modal = 'editItem';
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
});
