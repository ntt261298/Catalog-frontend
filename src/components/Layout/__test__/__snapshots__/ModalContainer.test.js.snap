// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Layout/ModalContainer.js should render addItem modal 1`] = `<LoadableComponent />`;

exports[`src/components/Layout/ModalContainer.js should render editItem modal 1`] = `<LoadableComponent />`;

exports[`src/components/Layout/ModalContainer.js should render login modal 1`] = `<LoadableComponent />`;

exports[`src/components/Layout/ModalContainer.js should render signup modal 1`] = `<LoadableComponent />`;
